docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,244 @@
+.. _custom-sockets:
+
+=========================
+Custom Sockets in Syncano
+=========================
+
+``Syncano`` gives its users an ability to create custom sockets. It means, that users can define 
+a very specific endpoints in their Syncano application, and use them as other Syncano 
+modules (Classes, Scripts, etc), using standard API calls.
+Currently, custom sockets allow only one dependency - Scripts. It means that on the backend side,
+each API call executes a Script and result of the execution, is returned as a result of the
+API call.
+
+Creating a custom socket
+------------------------
+
+To create a custom socket follow these steps::
+
+    import syncano
+    from syncano.models import CustomSocket, Endpoint, ScriptCall, ScriptDependency, RuntimeChoices
+    from syncano.connection import Connection
+
+    # 1. Initialize a custom socket.
+    custom_socket = CustomSocket(name='my_custom_socket')  # this will create an object in place (do API call)
+
+    # 2. Define endpoints.
+    my_endpoint = Endpoint(name='my_endpoint')  # no API call here
+    my_endpoint.add_call(ScriptCall(name='custom_script'), methods=['GET'])
+    my_endpoint.add_call(ScriptCall(name='another_custom_script'), methods=['POST'])
+
+    # Explanation of the above lines:
+    # Defined endpoint will be visible under `my_endpoint` name:
+    # On this syncano API endpoint the above endpoint will be called (after custom socket creation)
+    # <host>://<api_version>/instances/<instance_name>/endpoints/sockets/my_endpoint/
+    # On this syncano API endpoint the details of the defined endpoint will be returned
+    # <host>://<api_version>/instances/<instance_name>/sockets/my_custom_socket/endpoints/my_endpoint/
+    # For the above endpoint - the two calls are defined, one uses GET method - the custom_script will be executed
+    # there, second uses the POST method and then the another_custom_script will be called;
+    # Currently only script are available for calls;
```

Comment:
I don't really understand above comments. Could someone rewrite them in other words in here?